src/ucp/api/ucp.h

```diff
@@ -1841,7 +1901,7 @@ ucs_status_t ucp_request_test(void *request, ucp_tag_recv_info_t *info);
  * explicitly once it is completed or released and we should return an error if
  * the request is not in one of these states.
  */
-void ucp_request_release(void *request);
+void ucp_worker_request_release(ucp_worker_h worker, void *request);
```

Comment:
we may revert to the initial implementation of using the memory pool to track the worker.
but need to add UCS API such as ucs_mpool_obj_owner(void *obj) to get the owning mpool from an object